### ISSUE TITLE
[Bug] The Legend of Zelda: Breath of the Wild doesn't load into save with Hypervisor enabled

### ISSUE THREAD

**Issue by blackxfiied** [Ryujinx/Ryujinx#6727]
### Description of the issue

I made an issue a while back (#5219)
This was before macOS had been fully upstreamed.

It was resolved by #5898, and i've been hearing success stories.
But it still refuses to work on my MacBook Pro (14", 2021) 

It gets to the end of the loading screen after pressing continue and then crashes.

### Reproduction steps

Attempt to load into a save in Zelda BotW

### Log file

[Ryujinx_1.1.1101_2024-02-28_12-07-19.log](https://github.com/Ryujinx/Ryujinx/files/15132796/Ryujinx_1.1.1101_2024-02-28_12-07-19.log)

### OS

macOS 14.3.1

### Ryujinx version

1.1.1291

### Game version

1.6.0 w/ both DLC packs

### CPU

M1 Pro

### GPU

M1 Pro

### RAM

16GB

**Comment by gdkchan** [comment]
That change was reverted a while after. #6068

**Comment by blackxfiied** [comment]
Would it be possble to add a toggle below the hypervisor button in the UI?
It could be called 'Enable Memory Access Workaround' or something similar.
Kinda like how Graphics >> Enable Texture Recompression is implemented because of a select few games that use ASTC textures?

If so, can I create a new issue and possibly create a PR for it? (if i can figure out what's going on in the backend, that is 😅)

**Comment by GamerzHell9137** [comment]
> Would it be possble to add a toggle below the hypervisor button in the UI? It could be called 'Enable Memory Access Workaround' or something similar. Kinda like how Graphics >> Enable Texture Recompression is implemented because of a select few games that use ASTC textures?
> 
> If so, can I create a new issue and possibly create a PR for it? (if i can figure out what's going on in the backend, that is 😅)

They wouldn't do that since the code is causing issues. You can locally build it yourself tho by readding the code back. All you need to do is go to the PR and see what files were changed.

**Comment by MutantAura** [comment]
> Would it be possble to add a toggle below the hypervisor button in the UI? It could be called 'Enable Memory Access Workaround' or something similar. Kinda like how Graphics >> Enable Texture Recompression is implemented because of a select few games that use ASTC textures?

The game works fine if you just disable the hypervisor in settings. 
Adding this workaround back does not solve anything as the game still softlocked around 5-10 mins into gameplay.

**Comment by blackxfiied** [comment]
I do locally build it, but I'm talking about an upstream implementation.
Yeah, it works fine without the hypervisor, but for the sake of CPU overhead I'd like it on where possible.

+ The game softlocks while loading data. You can play normally until you attempt to load anything else other than where you are.

**Comment by asianmaster64** [comment]
Do you just fix the crashing on start by disabling hypervisor?

**Comment by blackxfiied** [comment]
> Do you just fix the crashing on start by disabling hypervisor?

Yeah, or alternatively you can merge the older commits that allow BotW to work using the hypervisor.

**Comment by asianmaster64** [comment]
I’m kind of noob and new to ryujinx 
What’s a commit? btw I turned of hypervisor and it still crashes right after ryujinx loads the game.

**Comment by blackxfiied** [comment]
a commit is just a dump of changes you make to code
say you add something to your hypothetical app
it changes two files in your source code
when you commit, you make a point of reference to where those two files were changed

find more here:
https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/about-commits
